### ISSUE TITLE
`minikube start` doesn't stop after download failure due to network issues

### ISSUE THREAD

**Issue by lxzhang000** [kubernetes/minikube#10164]
I'm testing the minikube network connection issues in China and happened to find there is a potential bug:

After all image fallback addresses failed, minikube didn't stop to the process as expected, and failed later due to lack of valid images.

**Steps to reproduce the issue:** 
os version: macOs catalina 10.15.7
minikube version: v1.16.0 commit: 9f1e482427589ff8451c4723b6ba53bb9742fbb1

minikube start --driver=virtualbox --v=1 --alsologtostderr

**Full output of failed command:**
I0114 22:46:46.233396   10344 cache.go:159] failed to download docker.pkg.github.com/kubernetes/minikube/kicbase:v0.0.15-snapshot4, will try fallback image if available: GET https://docker.pkg.github.com/v2/kubernetes/minikube/kicbase/manifests/v0.0.15-snapshot4: UNAUTHORIZED: GitHub Docker Registry needs login
**E0114 22:46:46.233427   10344 cache.go:180] Error downloading kic artifacts:  failed to download kic base image or any fallback image
I0114 22:46:46.233444   10344 cache.go:185] Successfully downloaded all kic artifacts**
I0114 22:46:46.233473   10344 start.go:314] acquiring machines lock for minikube: {Name:mkb987fd896c2845336bd726163c7ebf1c4ecf2e Clock:{} Delay:500ms Timeout:10m0s Cancel:<nil>}
I0114 22:46:46.233601   10344 start.go:318] acquired machines lock for "minikube" in 114.554µs
I0114 22:46:46.233634   10344 start.go:90] Provisioning new machine with config: &{Name:minikube KeepContext:false EmbedCerts:false MinikubeISO: KicBaseImage:registry.cn-hangzhou.aliyuncs.com/google_containers/kicbase:v0.0.15-snapshot4@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb3

I checked the master code:

In pkg/minikube/node/cache.go, in function "[beginDownloadKicBaseImage](https://github.com/kubernetes/minikube/blob/caf59caaea357eef1ac1811ec514d883297523b6/pkg/minikube/node/cache.go#L161)", if all fallback addresses fails, it returns:

`fmt.Errorf("failed to download kic base image or any fallback image")`

I think it should wrap the type of the error here, so the function "[waitDownloadKicBaseImage](https://github.com/kubernetes/minikube/blob/caf59caaea357eef1ac1811ec514d883297523b6/pkg/minikube/node/cache.go#L177)" could go to the desired branch and exit.

**Comment by sharifelgamal** [comment]
Yeah, I agree we could do better here and fail more gracefully. Help wanted!

**Comment by andriyDev** [comment]
@vishjain Are you still working on this?

**Comment by vishjain** [comment]
> @vishjain Are you still working on this?

Yes, I'm working on this one.

**Comment by andriyDev** [comment]
Awesome!

**Comment by medyagh** [comment]
> I'm testing the minikube network connection issues in China and happened to find there is a potential bug:
> 
> After all image fallback addresses failed, minikube didn't stop to the process as expected, and failed later due to lack of valid images.
> 
> **Steps to reproduce the issue:**
> os version: macOs catalina 10.15.7
> minikube version: v1.16.0 commit: [9f1e482](https://github.com/kubernetes/minikube/commit/9f1e482427589ff8451c4723b6ba53bb9742fbb1)
> 
> minikube start --driver=virtualbox --v=1 --alsologtostderr
> 
> **Full output of failed command:**
> I0114 22:46:46.233396 10344 cache.go:159] failed to download docker.pkg.github.com/kubernetes/minikube/kicbase:v0.0.15-snapshot4, will try fallback image if available: GET https://docker.pkg.github.com/v2/kubernetes/minikube/kicbase/manifests/v0.0.15-snapshot4: UNAUTHORIZED: GitHub Docker Registry needs login
> **E0114 22:46:46.233427 10344 cache.go:180] Error downloading kic artifacts: failed to download kic base image or any fallback image I0114 22:46:46.233444 10344 cache.go:185] Successfully downloaded all kic artifacts**
> I0114 22:46:46.233473 10344 start.go:314] acquiring machines lock for minikube: {Name:mkb987fd896c2845336bd726163c7ebf1c4ecf2e Clock:{} Delay:500ms Timeout:10m0s Cancel:}
> I0114 22:46:46.233601 10344 start.go:318] acquired machines lock for "minikube" in 114.554µs
> I0114 22:46:46.233634 10344 start.go:90] Provisioning new machine with config: &{Name:minikube KeepContext:false EmbedCerts:false MinikubeISO: KicBaseImage:registry.cn-hangzhou.aliyuncs.com/google_containers/kicbase:v0.0.15-snapshot4@sha256:ef1f485b5a1cfa4c989bc05e153f0a8525968ec999e242efff871cbb3
> 
> I checked the master code:
> 
> In pkg/minikube/node/cache.go, in function "[beginDownloadKicBaseImage](https://github.com/kubernetes/minikube/blob/caf59caaea357eef1ac1811ec514d883297523b6/pkg/minikube/node/cache.go#L161)", if all fallback addresses fails, it returns:
> 
> `fmt.Errorf("failed to download kic base image or any fallback image")`
> 
> I think it should wrap the type of the error here, so the function "[waitDownloadKicBaseImage](https://github.com/kubernetes/minikube/blob/caf59caaea357eef1ac1811ec514d883297523b6/pkg/minikube/node/cache.go#L177)" could go to the desired branch and exit.

@lxzhang000 this seems to be a very old minikube version ! and also this is virtualbox driver this should have nothing to do with KIC or docker driver

I am not convinceced what this issue is purposing to do
